### PR TITLE
ref(starlite): Use new scopes API

### PR DESCRIPTION
**Pull request by sentrivana**
---

## General Notes

Thank you for contributing to `sentry-python`!

Please add tests to validate your changes, and lint your code using `tox -e linters`.

Running the test suite on your PR might require maintainer approval. Some tests (AWS Lambda) additionally require a maintainer to add a special label to run and will fail if the label is not present.

#### For maintainers

Sensitive test suites require maintainer review to ensure that tests do not compromise our secrets. This review must be repeated after any code revisions.

Before running sensitive test suites, please carefully check the PR. Then, apply the `Trigger: tests using secrets` label. The label will be removed after any code changes to enforce our policy requiring maintainers to review all code revisions before running sensitive tests.

### CODE REVIEW

File: sentry_sdk/integrations/starlite.py [hunk 1]
```diff
@@ -114,51 +115,50 @@ def enable_span_for_middleware(middleware: "Middleware") -> "Middleware":
...
-                        span.set_tag("starlite.middleware_name", middleware_name)
-                        return await send(message)
-
-                send_name = getattr(send, "__name__", str(send))
-                send_patched = send_name == "_sentry_send"
-                new_send = _sentry_send if not send_patched else send
-
-                return await old_call(self, scope, new_receive, new_send)
-        else:
+        if sentry_sdk.get_client().get_integration(StarliteIntegration) is None:
```
Comment (antonpirker) [inline sentry_sdk/integrations/starlite.py:123]: We can use the @encure_... decorator here, no?

File: sentry_sdk/integrations/starlite.py [hunk 2]
```diff
@@ -176,66 +176,67 @@ def patch_http_route_handle() -> None:
     async def handle_wrapper(
         self: "HTTPRoute", scope: "HTTPScope", receive: "Receive", send: "Send"
     ) -> None:
-        hub = Hub.current
-        integration: StarliteIntegration = hub.get_integration(StarliteIntegration)
+        integration: StarliteIntegration = sentry_sdk.get_client().get_integration(
+            StarliteIntegration
+        )
         if integration is None:
             return await old_handle(self, scope, receive, send)
```
Comment (antonpirker) [inline sentry_sdk/integrations/starlite.py:183]: We could also use the @ensure_... decorator here.

File: sentry_sdk/integrations/starlite.py [hunk 3]
```diff
@@ -176,66 +176,67 @@ def patch_http_route_handle() -> None:
...
+            scope=scope, receive=receive, send=send
+        )
+        extracted_request_data = ConnectionDataExtractor(
+            parse_body=True, parse_query=True
+        )(request)
+        body = extracted_request_data.pop("body")
+
+        request_data = await body
+
+        def event_processor(event: "Event", _: "Dict[str, Any]") -> "Event":
```
Comment (antonpirker) [inline sentry_sdk/integrations/starlite.py:196]: Instead of `"Dict[str, Any]"` we can use the `sentry_sdk.types.Hint` type

**Review by antonpirker — COMMENTED** [review]
Some questions

File: sentry_sdk/integrations/starlite.py (hunk 3 above)
Comment (sentrivana) [reply sentry_sdk/integrations/starlite.py]: Nice!

File: sentry_sdk/integrations/starlite.py (hunk 1 above)
Comment (sentrivana) [reply sentry_sdk/integrations/starlite.py:124]: I'd do this after we've fixed it and we're sure it works as intended, then we can go over the async integrations and add it.

**Review by szokeasaurusrex — APPROVED** [review]

**Pull request merged**